src/menu_events.cpp

```diff
@@ -454,6 +454,29 @@ void menu_handler::show_enemy_moves(bool ignore_units, int side_num)
 			gui_->highlight_another_reach(path);
 		}
 	}
+	
+	//   Fetch cursor position, in order to later find a possible
+	// unit under it.
+	int x = -1;
+	int y = -1;
+	// auto bitmask =
+			SDL_GetMouseState(&x, &y);
+
+	//   Find possible unit (no matter whether friend or foe) under
+	// the cursor.
+	const mouse_handler& mh = const_cast<mouse_handler&>(
+			pc_.get_mouse_handler_base());
+	const map_location hex_under_mouse = mh.hex_clicked_on(x, y);
+	const auto selected_hex_has_unit = mh.hex_hosts_unit(hex_under_mouse);
+    
+	if (selected_hex_has_unit) {
+		//   At this point, a single pixel move would remove the
+		// enemy [best] movements highlight hex tiles, so some
+		// prevention on normal unit mouseover movement highlight
+		// has to be toggled.
+
+		const_cast<mouse_handler&>(mh).prevent_units_highlight();
```

Comment:
Remove `const_cast`; `mh` should be a non-const reference.

Reply:
We already discussed that a little, take a look at my comments on the `const_cast` and his responses.

Reply:
My intention was to be `const` correct, even if that required considering a slightly pedantic interpretation of `const` correctness; but after yet another pass reading the [coding standards](https://wiki.wesnoth.org/CodingStandards) I think the request is right and will be trying to get the request complied.